pkg/worker/workerhandler/claimresolvehandler/custom_ensure.go

```diff
@@ -17,38 +19,83 @@ import (
 func (h *SystemHandler) Ensure(tas *task.Task, bud *budget.Budget) error {
 	var err error
 
-	//     find claims with lifecycle "propose" that are expired
-	var cla []*poststorage.Object
+	// find trees with lifecycle "propose" that are expired
+	var trees []*poststorage.Object
 	{
-		cla, err = h.sto.Post().SearchExpiry()
+		trees, err = h.sto.Post().SearchExpiry()
 		if err != nil {
 			return tracer.Mask(err)
 		}
 	}
 
-	for _, x := range cla {
-		var tre *big.Int
+	for _, x := range trees {
+		var treeId *big.Int
 		{
-			tre, err = metToTre(x.Meta)
+			treeId, err = metaToTreeId(x.Meta)
 			if err != nil {
 				return tracer.Mask(err)
 			}
 		}
 
-		var det [4]marketscontract.IMarketsClaim
+		var claims [4]marketscontract.IMarketsClaim
 		{
-			det, err = h.mar.Claims(nil, tre)
+			claims, err = h.markets.Claims(nil, treeId)
 			if err != nil {
 				return tracer.Mask(err)
 			}
 		}
 
-		fmt.Printf("det %#v\n", det)
+		var claimsLength *big.Int
+		{
+			claimsLength, err = h.markets.ClaimsLength(nil, treeId)
+			if err != nil {
+				return tracer.Mask(err)
+			}
 
-		// read from chain if claim has to be resolved, just double checking
+			// sanity check - we cap the number of claims to 4 in the contract
+			if claimsLength.Cmp(big.NewInt(4)) == 1 {
+				return tracer.Maskf(runtime.ExecutionFailedError, "claimresolverhandler: invalid state: too many claims")
+			}
+		}
 
-		// 1. claim is not yet resolvable, this is a bug, we just log an error,
-		// maybe send an email
+		var claim marketscontract.IMarketsClaim
+		{
+			claim = claims[claimsLength.Int64()-1]
+		}
+
+		if claim.Status == 1 { // claim.Status == Active
+
+			var yeaVoters []string
+			var nayVoters []string
+			yeaStakersLength := len(claim.Stake.YeaStakers)
+			nayStakersLength := len(claim.Stake.NayStakers)
+			if yeaStakersLength > 0 && nayStakersLength > 0 {
+				votersLength := int(math.Min(float64(yeaStakersLength), float64(nayStakersLength)))
+				for i := 0; i < votersLength; i++ {
+					randomIndex := rand.Intn(yeaStakersLength)
+					yeaVoters = append(yeaVoters, claim.Stake.YeaStakers[randomIndex].Hex())
+
+					randomIndex = rand.Intn(nayStakersLength)
+					nayVoters = append(nayVoters, claim.Stake.NayStakers[randomIndex].Hex())
+				}
+			} else if yeaStakersLength > 0 && nayStakersLength == 0 {
+				randomIndex := rand.Intn(yeaStakersLength)
+				yeaVoters = append(yeaVoters, claim.Stake.YeaStakers[randomIndex].Hex())
+			} else if nayStakersLength > 0 && yeaStakersLength == 0 {
+				randomIndex := rand.Intn(nayStakersLength)
+				nayVoters = append(nayVoters, claim.Stake.NayStakers[randomIndex].Hex())
+			} else {
+				return tracer.Maskf(runtime.ExecutionFailedError, "claimresolverhandler: invalid state: invalid stakers")
+			}
```

Comment:
Let's move this to a separate function that can be properly unit tested. 

* We want to know how many random addresses have been selected.
* We want to cover different cases of situations where more or less stakers are present on either side.
* We need to make sure that there are no duplicates in the result. 
* We could also test that the results are really randomish somehow. (bonus points)